string text
int32 n
float32 f